homeassistant/components/airly/__init__.py

```diff
@@ -49,43 +53,75 @@
 
     websession = async_get_clientsession(hass)
 
-    airly = AirlyData(websession, api_key, latitude, longitude)
+    # We check how many Airly config entries are and calculate interval to not
+    # exceed allowed numbers of requests.
+    update_interval = timedelta(
+        minutes=ceil(24 * 60 / MAX_REQUESTS_PER_DAY) * instances(hass)
+    )
+
+    coordinator = AirlyDataUpdateCoordinator(
+        hass, websession, api_key, latitude, longitude, update_interval
+    )
+    await coordinator.async_refresh()
 
-    await airly.async_update()
+    hass.data.setdefault(DOMAIN, {})
+    hass.data[DOMAIN][config_entry.entry_id] = coordinator
 
-    hass.data[DOMAIN][DATA_CLIENT][config_entry.entry_id] = airly
+    for component in PLATFORMS:
+        hass.async_create_task(
+            hass.config_entries.async_forward_entry_setup(config_entry, component)
+        )
+
+    # Change update_interval for another Airly instances
+    for entry in hass.config_entries.async_entries(DOMAIN):
```

Comment:
Why go through here and not just directly iterate over `hass.data[DOMAIN].values()` ?

Reply:
Very good question ;) Fixed.